app/models/start_teacher_training.rb

```diff
@@ -1,13 +1,18 @@
 class StartTeacherTraining < Base
   attribute :year_of_entry, :string
 
+  validates :year_of_entry, types: { method: :get_candidate_initial_teacher_training_years, message: "You must select an option from the list" }
   validate :date_cannot_be_in_the_past, unless: :dont_know
 
-  def year_range(number_of_years) # sets year range for view
-    years = (Date.today.year..Date.today.next_year(number_of_years).year).map do |year|
-      OpenStruct.new(value: year, name: year)
-    end
-    years << OpenStruct.new(value: "don't know", name: "Don't know")
+  def self.options
+    option_list = {}
```

Comment:
Take it or leave it:
```
ApiClient::get_candidate_initial_teacher_training_years.reduce({}) { |options, year| options[type.value] = type.id }
```

Reply:
Coolio, I think this works:
ApiClient::get_candidate_initial_teacher_training_years.reduce({}) { |options, year| options.update(year.value => year.id) }